src/Symfony/Component/CssSelector/Parser/Tokenizer/TokenizerPatterns.php

```diff
@@ -43,7 +43,7 @@ public function __construct()
         $this->stringEscapePattern = $this->newLineEscapePattern.'|'.$this->escapePattern;
         $this->nonAsciiPattern = '[^\x00-\x7F]';
         $this->nmCharPattern = '[_a-z0-9-]|'.$this->escapePattern.'|'.$this->nonAsciiPattern;
-        $this->nmStartPattern = '[_a-z]|'.$this->escapePattern.'|'.$this->nonAsciiPattern;
+        $this->nmStartPattern = '[_a-z-]|'.$this->escapePattern.'|'.$this->nonAsciiPattern;
```

Comment:
Checking the [official CSS grammar](https://www.w3.org/TR/CSS21/grammar.html#scanner) I found this:

```
nmstart		[_a-z]|{nonascii}|{escape}
nmchar		[_a-z0-9-]|{nonascii}|{escape}
ident		-?{nmstart}{nmchar}*
```

So maybe we should move the optional starting `-` to `$this->identifierPattern`?

Reply:
Sure, fixed according to CSS grammar.